src/network/networkd-link.c

```diff
@@ -640,6 +656,12 @@ static int link_stop_clients(Link *link) {
                         r = log_link_warning_errno(link, k, "Could not stop IPv6 Router Discovery: %m");
         }
 
+        if (link->radv) {
+                k = sd_radv_stop(link->radv);
+                if (k < 0)
+                        r = log_link_warning_errno(link, k, "Could not stop IPv6 Router Advertisment: %m");
```

Comment:
Not something to fix in this patch, but maybe some other time: it would be better to return the first error, not the last one.